python/lsst/cp/pipe/defects.py

```diff
@@ -367,27 +393,91 @@ def _writeData(self, dataRef, defects):
         defects : `lsst.meas.algorithms.Defect`
             The defects to be written.
         """
-        filename = dataRef.getUri(write=True)  # does not guarantee that full path exists
-        dirname = os.path.dirname(filename)
-        if not os.path.exists(dirname):
-            os.makedirs(dirname)
+        date = str(dafBase.DateTime(midTime, dafBase.DateTime.MJD).toPython())
+        date = date.replace(" ", "T")  # original has a space in the middle
+
+        detName = self._getDetectorName(dataRef)
+        instrumentName = self._getInstrumentName(dataRef)
+        detNum = self._getDetectorNumber(dataRef)
+        if not self.config.ignoreFilters:
+            filt = self._filterSetToFilterString(filters)
+        else:
+            filt = self.config.nullFilterName
+
+        CALIB_ID = f"detectorName={detName} detector={detNum} calibDate={date} ccd={detNum} filter={filt}"
+        try:
+            raftName = detName.split("_")[0]
+            CALIB_ID += f" raftName={raftName}"
+        except Exception:
+            pass
+
+        now = dafBase.DateTime.now().toPython()
+        mdOriginal = defects.getMetadata()
+        mdSupplemental = {"INSTRUME": instrumentName,
+                          "DETECTOR": dataRef.dataId['detector'],
+                          "CALIBDATE": date,
+                          "CALIB_ID": CALIB_ID,
+                          "CALIB_CREATION_DATE": now.date().isoformat(),
+                          "CALIB_CREATION_TIME": now.time().isoformat()}
+
+        for k, v in mdSupplemental.items():
+            mdOriginal.add(k, v)
+        defects.setMetadata(mdOriginal)
+
+        templateFilename = dataRef.getUri(write=True)  # does not guarantee that full path exists
```

Comment:
This all seems wrong if butler is involved. Why can't butler.put work directly? Why do you have to ask butler for a template at all?

Reply:
Come chat with me and KT as per Slack to resolve this 🙂

Reply:
Discussion OOB has resolved this, I think. I am putting in a TODO to sort this, and the comment below, out in Gen3 and leaving as is for now.